cg/store/api/status.py

```diff
@@ -594,19 +595,27 @@ def get_analyses_before_date(
         records = records.filter(models.Analysis.started_at <= before)
         return records
 
-    def observations_to_upload(self):
-        """Fetch observations that haven't been uploaded."""
-
-        return self.Family.query.join(
+    def observations_to_upload(self, pipeline: Pipeline = None) -> Query:
+        """Fetch observations that have not been uploaded."""
+        records: Query = self.Family.query.join(
```

Comment:
Could you split this query into a separate unit tested function for future reuse? Similar to: get_families_with_analyses. It is not yet "locked" to loquDB - that happens in the filtering so we should be able to reuse it elsewhere.

Reply:
Yes, definetly!